cms/search/wagtail_hooks.py

```diff
@@ -0,0 +1,33 @@
+from typing import TYPE_CHECKING
+
+from wagtail import hooks
+
+from cms.settings.base import EXCLUDED_PAGE_TYPES
+
+from . import get_publisher
+
+if TYPE_CHECKING:
+    from django.http import HttpRequest
+    from wagtail.models import Page
+
+publisher = get_publisher()
+
+
+@hooks.register("after_publish_page")
+def page_published(request: "HttpRequest", page: "Page") -> None:
+    """This hook is called after a page is published, either via the Wagtail admin UI or programmatically.
+
+    Wagtail hook documentation: https://docs.wagtail.io/en/stable/reference/hooks.html#after-publish-page
```

Comment:
The Wagtail hooks are called only for actions via the UI.

Please use the corresponding signals:
- https://docs.wagtail.org/en/stable/reference/signals.html#page-published and 
- https://docs.wagtail.org/en/stable/reference/signals.html#page-unpublished
- and https://docs.djangoproject.com/en/5.1/ref/signals/#post-delete for page deletion

Reply:
Removed the use of Wagtail hooks, using Signals now.